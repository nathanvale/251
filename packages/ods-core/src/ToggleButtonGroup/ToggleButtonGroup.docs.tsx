import React from "react";

import { ComponentDocs } from "@origin-digital/ods-types";
import {
  ToggleButtonGroup,
  ToggleButtonGroupProps,
} from "../ToggleButtonGroup/ToggleButtonGroup";
import { Text } from "../Text";
import { Stack } from "../Stack";
import {
  generateToggleButtonGroup,
  toggleButtons,
  toggleButtonsObject,
} from "./ToggleButtonGroup.helper";

export const docs: ComponentDocs<ToggleButtonGroupProps> = {
  category: "Interaction",
  componentName: "ToggleButtonGroup",
  description: [
    "This component is similar to a RadioGroup. It allows selection of 1 out of all options.",
    "This component could be use either Uncontrolled (if defaultValue is set) or Controlled (if its value) is set.",
  ].join(" "),
  propDescriptions: {
    className: `The CSS classname to be applied to the group component.`,
    "aria-label": `Use this to provide a mre useful label for the group which will be used by screen readers`,
    children: [
      `Use this to manually render the child ToggleButton components.`,
      `If provided, the "options" prop will be ignored.`,
      `You can use this to have full control over how the toggle buttons are rendered.`,
    ].join(" "),
    "data-id": `USe this to provide an id for test purposes`,
    defaultValue: `In case of uncontrolled ToggleButtonGroup, this can be used to provide the initial value`,
    id: `The unique id of the ToggleButtonGroup, required for accessibility.`,
    onChange: [
      `Is called every time the selected ToggleButton is changed.`,
      `To read the new value, use either "event.target.value" or "value" (the 2nd argument).`,
    ].join(" "),
    options: [
      `The list of ToggleButton values. This can have two formats. If it is of type "string[]"`,
      `for each string it will render a ToggleButton with that text being the ToggleButton's child.`,
      `If it is of the form "{label: string; value: string | number}[]" then again for each object`,
      `in the array it renders a ToggleButton with its value being the value of the object and its child`,
      `being the label of the object.`,
    ].join(" "),
    size: `Defines the height of its child ToggleButtons`,
    value: [
      `If passed in, makes ToggleButtonGroup a controlled component.`,
      `It will also automatically select the corresponding ToggleButton.`,
      `If this value is unset initially, the ToggleButtonGroup will be Uncontrolled.`,
    ].join(" "),
    width: [
      `By defualt it is "full" which means the ToggleButtonGroup will take up 100% width of its parent`,
      `and the widths of all child ToggleButtons will be equal.`,
      `If it is "content" each child ToggleButton takes up the width based on its child content and`,
      `as a result each ToggleButton will of a different width.`,
    ].join(" "),
  },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => generateToggleButtonGroup({ defaultValue: toggleButtons[2] }),
    },
    additional: [
      {
        label: "Uncontrolled - setting defaultValue",
        Code: () =>
          generateToggleButtonGroup({ defaultValue: toggleButtons[1] }),
      },
      {
        label: "Controlled, passing down value and onChange props",
        playroom: false,
        Code: () => {
          const [value, setValue] = React.useState<string>(toggleButtons[2]);
          return (
            <Stack space="small">
              <Text weight="medium" tone="positive">
                value: {value}
              </Text>
              <ToggleButtonGroup
                value={value}
                id="Australian city"
                options={toggleButtons}
                onChange={(_, v) => setValue(v)}
              />
            </Stack>
          );
        },
        codeString: `const [value, setValue] = React.useState<string>("${
          toggleButtons[3]
        }");
return <Stack space="small">
  <Text weight="medium" tone="positive">
    value: {value}
  </Text>
  <ToggleButtonGroup
    value={value}
    id="Australian city"
    onChange={(_, val) => setValue(val)}
    options={${JSON.stringify(toggleButtons, null, 2)}}
  />
</Stack>;`,
      },
      {
        label: "Different (label, value) pair for options",
        Code: () =>
          generateToggleButtonGroup(
            {
              defaultValue: toggleButtonsObject[2].value,
            },
            "object"
          ),
      },
      {
        label: "Custom ToggleButtons via children prop",
        Code: () =>
          generateToggleButtonGroup(
            {
              defaultValue: toggleButtonsObject[1].value,
            },
            "ReactNode"
          ),
      },
      {
        label: "Content width, each ToggleButton has the size of its content",
        Code: () =>
          generateToggleButtonGroup({
            defaultValue: toggleButtons[1],
            width: "content",
          }),
      },
      {
        label: "disabled ToggleButtons",
        Code: () =>
          generateToggleButtonGroup(
            {
              defaultValue: toggleButtonsObject[1].value,
            },
            "ReactNode",
            2
          ),
      },
      {
        label: "Different sizes",
        Code: () => (
          <Stack space="small">
            Small
            {generateToggleButtonGroup({
              defaultValue: toggleButtons[1],
              size: "small",
            })}
            Medium
            {generateToggleButtonGroup({
              defaultValue: toggleButtons[1],
            })}
          </Stack>
        ),
      },
    ],
  },
  snippets: [
    {
      label: "Default",
      Code: () => generateToggleButtonGroup({ defaultValue: toggleButtons[2] }),
    },
    {
      label: "(label, value) pair",
      Code: () =>
        generateToggleButtonGroup(
          { defaultValue: toggleButtonsObject[1].value },
          "object"
        ),
    },
    {
      label: "Explicit children",
      Code: () =>
        generateToggleButtonGroup(
          { defaultValue: toggleButtonsObject[3].value },
          "ReactNode"
        ),
    },
  ],
};
