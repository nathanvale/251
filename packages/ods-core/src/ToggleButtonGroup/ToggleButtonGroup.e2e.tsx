import { E2ETests } from "@origin-digital/ods-types";

import {
  generateToggleButtonGroup,
  toggleButtons,
  toggleButtonsObject,
} from "./ToggleButtonGroup.helper";

export const tests: E2ETests = [
  {
    label: "Default ToggleButtonGroup",
    Code: () => generateToggleButtonGroup({ defaultValue: toggleButtons[2] }),
  },
  {
    label: "Content width",
    Code: () =>
      generateToggleButtonGroup({
        defaultValue: toggleButtons[1],
        width: "content",
      }),
  },
  {
    label: "Disabled ToggleButton",
    Code: () =>
      generateToggleButtonGroup(
        {
          defaultValue: toggleButtonsObject[1].value,
        },
        "ReactNode",
        2
      ),
  },
  {
    label: "Small ToggleButtonGroup",
    Code: () =>
      generateToggleButtonGroup({
        defaultValue: toggleButtons[1],
        size: "small",
      }),
  },
];
