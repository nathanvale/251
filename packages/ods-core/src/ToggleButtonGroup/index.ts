export { ToggleButtonGroup } from "./ToggleButtonGroup";
export type ToggleButtonGroupProps = import("./ToggleButtonGroup").ToggleButtonGroupProps;
