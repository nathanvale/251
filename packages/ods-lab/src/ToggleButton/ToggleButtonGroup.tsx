import * as React from "react";
import MuiToggleButtonGroup, {
  ToggleButtonGroupProps as IToggleButtonGroup,
} from "@material-ui/lab/ToggleButtonGroup";

export interface ToggleButtonGroupProps extends IToggleButtonGroup {}

export const ToggleButtonGroup = (props: ToggleButtonGroupProps) => (
  <MuiToggleButtonGroup {...props}>{props.children}</MuiToggleButtonGroup>
);
