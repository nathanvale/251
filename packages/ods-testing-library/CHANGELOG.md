# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
