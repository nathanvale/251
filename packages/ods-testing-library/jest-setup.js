import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
