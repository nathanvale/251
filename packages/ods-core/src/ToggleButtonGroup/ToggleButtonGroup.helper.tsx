import React from "react";
import { ToggleButton } from "../ToggleButton/ToggleButton";
import { ToggleButtonGroup, ToggleButtonGroupProps } from "./ToggleButtonGroup";

export const toggleButtons = ["Sydney", "Brisbane", "Perth", "Melbourne"];
export const toggleButtonsObject = [
  { label: "Sydney", value: "sydn" },
  { label: "Brisbane", value: "brsb" },
  { label: "Perth", value: "prth" },
  { label: "Melbourne", value: "melb" },
];

export const generateToggleButtonGroup = (
  props: Partial<ToggleButtonGroupProps> = {},
  childrenType: "string" | "object" | "ReactNode" = "string",
  disabledIndex: number = -1
) => {
  const { id, "data-id": dataId, ...others } = props;
  const calcId = id || "Australian city";
  return (
    <ToggleButtonGroup
      data-id={dataId || calcId}
      id={calcId}
      {...(childrenType === "string"
        ? {
            options: toggleButtons,
          }
        : {})}
      {...(childrenType === "object"
        ? {
            options: toggleButtonsObject,
          }
        : {})}
      {...others}
    >
      {childrenType === "ReactNode" &&
        toggleButtonsObject.map((obj, idx) => (
          <ToggleButton
            aria-label={obj.label}
            key={obj.value}
            value={obj.value}
            {...(idx === disabledIndex ? { disabled: true } : {})}
          >
            {obj.label}
          </ToggleButton>
        ))}
    </ToggleButtonGroup>
  );
};
