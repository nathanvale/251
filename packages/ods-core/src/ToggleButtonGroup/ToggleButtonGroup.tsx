import React from "react";

import {
  ToggleButtonGroup as MuiToggleButtonGroup,
  ToggleButtonGroupProps as MuiToggleButtonGroupProps,
} from "@material-ui/lab";

import { ComponentBaseProps } from "@origin-digital/ods-types";
import { makeStyles } from "@material-ui/core/styles";
import { useControlled } from "@material-ui/core";
import { ToggleButton } from "../ToggleButton/ToggleButton";

export type ToggleButtonWidth = "full" | "content";
export type OptionType = { value: string | number; label: string };
export interface ToggleButtonGroupProps
  extends Omit<ComponentBaseProps, "disabled" | "id"> {
  id: string;
  "aria-label"?: string;
  defaultValue?: string | number;
  muiProps?: MuiToggleButtonGroupProps;
  onChange?: (event: React.MouseEvent<HTMLElement>, value: any) => void;
  options?: string[] | OptionType[];
  size?: "small" | "medium";
  value?: string | number;
  width?: ToggleButtonWidth;
}

const defaultProps: Partial<ToggleButtonGroupProps> = {
  width: "full",
};

const useTBGStyles = makeStyles(
  (_theme) => ({
    root: ({ width }: Partial<ToggleButtonGroupProps>) =>
      width === "full"
        ? {
            width: `100%`,
          }
        : {},
    grouped: ({ children, options, width }: Partial<ToggleButtonGroupProps>) =>
      width === "full" &&
      (children || (Array.isArray(options) && options.length))
        ? {
            width: `${
              100 /
              (children ? React.Children.count(children) : options?.length ?? 1)
            }%`,
          }
        : {},
  }),
  { classNamePrefix: "toggle-button-group" }
);

/**
 * If children provided it has precendence over options.
 * If not, render options based on whether it is a string[] or {value, label}[].
 */
const renderChildren = ({
  children,
  classes,
  options,
}: Pick<ToggleButtonGroupProps, "children" | "options" | "classes">) => {
  if (children) {
    return children;
  }
  if (Array.isArray(options) && options.length) {
    if (typeof options[0] === "string") {
      return (options as string[]).map((opt: string) => (
        <ToggleButton
          aria-label={opt}
          className={classes?.grouped}
          key={opt}
          value={opt}
        >
          {opt}
        </ToggleButton>
      ));
    } else if (
      Object.keys(options[0]).includes("value") &&
      Object.keys(options[0]).includes("label")
    ) {
      return (options as OptionType[]).map((opt: OptionType) => (
        <ToggleButton
          aria-label={opt.label}
          className={classes?.grouped}
          key={opt.value}
          value={opt.value}
        >
          {opt.label}
        </ToggleButton>
      ));
    }
    return null;
  }
  return null;
};

export const ToggleButtonGroup = ({
  "aria-label": ariaLabel,
  children,
  "data-id": dataId,
  defaultValue,
  id,
  muiProps,
  onChange,
  options,
  value: valueProp,
  width,
  ...others
}: ToggleButtonGroupProps) => {
  const calcDataId = dataId || id;
  const classes = useTBGStyles({ children, options, width });

  const [value, setValue] = useControlled({
    controlled: valueProp,
    default: defaultValue,
    name: "ToggleButtonGroup",
    state: "value",
  });

  const childrenToRender = React.useMemo(
    () => renderChildren({ children, classes, options }),
    [children, classes, options]
  );

  // Here we prevent deselceting all elements by not calling onChange if value is null.
  // value === null means the user has clicked the selected toggle.
  const handleChange = React.useMemo(
    () => (event: React.MouseEvent<HTMLElement>, value: any) => {
      if (value !== null) {
        setValue(value);
        onChange?.(event, value);
      }
    },
    [onChange, setValue]
  );
  return (
    <MuiToggleButtonGroup
      classes={classes}
      data-id={calcDataId}
      id={id}
      aria-label={ariaLabel || id}
      exclusive={true}
      onChange={handleChange}
      value={value}
      {...others}
      {...muiProps}
    >
      {childrenToRender}
    </MuiToggleButtonGroup>
  );
};

ToggleButtonGroup.defaultProps = defaultProps;

ToggleButtonGroup.displayName = "ToggleButtonGroup";
