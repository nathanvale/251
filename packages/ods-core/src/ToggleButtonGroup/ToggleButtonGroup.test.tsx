import { render, fireEvent } from "@origin-digital/ods-testing-library";
import { ToggleButtonGroupProps } from "./ToggleButtonGroup";
import {
  generateToggleButtonGroup,
  toggleButtons,
  toggleButtonsObject,
} from "./ToggleButtonGroup.helper";

const renderToggleButtonGroup = (
  props: Partial<ToggleButtonGroupProps> = {},
  childrenType: "string" | "object" | "ReactNode" = "string",
  disabledIndex: number = -1
) => render(generateToggleButtonGroup(props, childrenType, disabledIndex));

test("It renders with string[] options", () => {
  const { container } = renderToggleButtonGroup({
    defaultValue: toggleButtons[2],
  });
  expect(container).toMatchSnapshot();
});

test("It renders with object[] options", () => {
  const { container } = renderToggleButtonGroup(
    {
      defaultValue: toggleButtonsObject[2].value,
    },
    "object"
  );
  expect(container).toMatchSnapshot();
});

test("It renders with direct ToggleButton children", () => {
  const { container } = renderToggleButtonGroup(
    {
      defaultValue: toggleButtonsObject[1].value,
    },
    "ReactNode"
  );
  expect(container).toMatchSnapshot();
});

test("It renders with content width", () => {
  const { container } = renderToggleButtonGroup({
    defaultValue: toggleButtons[1],
    width: "content",
  });
  expect(container).toMatchSnapshot();
});

test("It renders disabled buttons", () => {
  const { container } = renderToggleButtonGroup(
    {
      defaultValue: toggleButtonsObject[1].value,
    },
    "ReactNode",
    2
  );
  expect(container).toMatchSnapshot();
});

test("It renders small buttons", () => {
  const { container } = renderToggleButtonGroup({
    defaultValue: toggleButtons[1],
    size: "small",
  });
  expect(container).toMatchSnapshot();
});

test("It sets the data-id", () => {
  const { getAllByTestId } = renderToggleButtonGroup({
    "data-id": "tbg-id",
    defaultValue: toggleButtons[1],
  });
  expect(getAllByTestId("tbg-id")).toHaveLength(1);
});

test("It sets aria-label to id", () => {
  const { getByTestId } = renderToggleButtonGroup({
    defaultValue: toggleButtons[1],
    id: "tbg-id",
  });
  expect(getByTestId("tbg-id")).toHaveAttribute("id", "tbg-id");
  expect(getByTestId("tbg-id")).toHaveAttribute("aria-label", "tbg-id");
});

test("It calls onChange handler", () => {
  const onChange = jest.fn();

  const { getByTestId } = renderToggleButtonGroup({
    defaultValue: toggleButtons[1],
    id: "my-tbg",
    onChange,
  });
  fireEvent.click(getByTestId("Melbourne"));
  expect(onChange).toHaveBeenCalled();
});
