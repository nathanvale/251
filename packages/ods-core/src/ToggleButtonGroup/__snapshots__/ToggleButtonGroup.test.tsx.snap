// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders disabled buttons 1`] = `
<div>
  <div
    aria-label="Australian city"
    class="MuiToggleButtonGroup-root toggle-button-group-root-29 toggle-button-group-root-31"
    data-id="Australian city"
    id="Australian city"
    role="group"
  >
    <button
      aria-label="Sydney"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-33 MuiToggleButtonGroup-grouped toggle-button-group-grouped-30 toggle-button-group-grouped-32 MuiToggleButtonGroup-groupedHorizontal"
      data-id="sydn"
      tabindex="0"
      type="button"
      value="sydn"
    >
      <span
        class="MuiToggleButton-label"
      >
        Sydney
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Brisbane"
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-33 MuiToggleButtonGroup-grouped toggle-button-group-grouped-30 toggle-button-group-grouped-32 MuiToggleButtonGroup-groupedHorizontal Mui-selected toggle-button-selected-34"
      data-id="brsb"
      tabindex="0"
      type="button"
      value="brsb"
    >
      <span
        class="MuiToggleButton-label"
      >
        Brisbane
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Perth"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-33 MuiToggleButtonGroup-grouped toggle-button-group-grouped-30 toggle-button-group-grouped-32 MuiToggleButtonGroup-groupedHorizontal Mui-disabled toggle-button-disabled-35 Mui-disabled"
      data-id="prth"
      disabled=""
      tabindex="-1"
      type="button"
      value="prth"
    >
      <span
        class="MuiToggleButton-label"
      >
        Perth
      </span>
    </button>
    <button
      aria-label="Melbourne"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-33 MuiToggleButtonGroup-grouped toggle-button-group-grouped-30 toggle-button-group-grouped-32 MuiToggleButtonGroup-groupedHorizontal"
      data-id="melb"
      tabindex="0"
      type="button"
      value="melb"
    >
      <span
        class="MuiToggleButton-label"
      >
        Melbourne
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`It renders small buttons 1`] = `
<div>
  <div
    aria-label="Australian city"
    class="MuiToggleButtonGroup-root toggle-button-group-root-36 toggle-button-group-root-38"
    data-id="Australian city"
    id="Australian city"
    role="group"
  >
    <button
      aria-label="Sydney"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-40 MuiToggleButtonGroup-grouped toggle-button-group-grouped-37 toggle-button-group-grouped-39 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-37 toggle-button-group-grouped-39 MuiToggleButton-sizeSmall"
      data-id="Sydney"
      tabindex="0"
      type="button"
      value="Sydney"
    >
      <span
        class="MuiToggleButton-label"
      >
        Sydney
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Brisbane"
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-40 MuiToggleButtonGroup-grouped toggle-button-group-grouped-37 toggle-button-group-grouped-39 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-37 toggle-button-group-grouped-39 Mui-selected toggle-button-selected-41 MuiToggleButton-sizeSmall"
      data-id="Brisbane"
      tabindex="0"
      type="button"
      value="Brisbane"
    >
      <span
        class="MuiToggleButton-label"
      >
        Brisbane
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Perth"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-40 MuiToggleButtonGroup-grouped toggle-button-group-grouped-37 toggle-button-group-grouped-39 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-37 toggle-button-group-grouped-39 MuiToggleButton-sizeSmall"
      data-id="Perth"
      tabindex="0"
      type="button"
      value="Perth"
    >
      <span
        class="MuiToggleButton-label"
      >
        Perth
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Melbourne"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-40 MuiToggleButtonGroup-grouped toggle-button-group-grouped-37 toggle-button-group-grouped-39 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-37 toggle-button-group-grouped-39 MuiToggleButton-sizeSmall"
      data-id="Melbourne"
      tabindex="0"
      type="button"
      value="Melbourne"
    >
      <span
        class="MuiToggleButton-label"
      >
        Melbourne
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`It renders with content width 1`] = `
<div>
  <div
    aria-label="Australian city"
    class="MuiToggleButtonGroup-root toggle-button-group-root-22 toggle-button-group-root-24"
    data-id="Australian city"
    id="Australian city"
    role="group"
  >
    <button
      aria-label="Sydney"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-26 MuiToggleButtonGroup-grouped toggle-button-group-grouped-23 toggle-button-group-grouped-25 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-23 toggle-button-group-grouped-25"
      data-id="Sydney"
      tabindex="0"
      type="button"
      value="Sydney"
    >
      <span
        class="MuiToggleButton-label"
      >
        Sydney
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Brisbane"
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-26 MuiToggleButtonGroup-grouped toggle-button-group-grouped-23 toggle-button-group-grouped-25 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-23 toggle-button-group-grouped-25 Mui-selected toggle-button-selected-27"
      data-id="Brisbane"
      tabindex="0"
      type="button"
      value="Brisbane"
    >
      <span
        class="MuiToggleButton-label"
      >
        Brisbane
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Perth"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-26 MuiToggleButtonGroup-grouped toggle-button-group-grouped-23 toggle-button-group-grouped-25 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-23 toggle-button-group-grouped-25"
      data-id="Perth"
      tabindex="0"
      type="button"
      value="Perth"
    >
      <span
        class="MuiToggleButton-label"
      >
        Perth
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Melbourne"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-26 MuiToggleButtonGroup-grouped toggle-button-group-grouped-23 toggle-button-group-grouped-25 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-23 toggle-button-group-grouped-25"
      data-id="Melbourne"
      tabindex="0"
      type="button"
      value="Melbourne"
    >
      <span
        class="MuiToggleButton-label"
      >
        Melbourne
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`It renders with direct ToggleButton children 1`] = `
<div>
  <div
    aria-label="Australian city"
    class="MuiToggleButtonGroup-root toggle-button-group-root-15 toggle-button-group-root-17"
    data-id="Australian city"
    id="Australian city"
    role="group"
  >
    <button
      aria-label="Sydney"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-19 MuiToggleButtonGroup-grouped toggle-button-group-grouped-16 toggle-button-group-grouped-18 MuiToggleButtonGroup-groupedHorizontal"
      data-id="sydn"
      tabindex="0"
      type="button"
      value="sydn"
    >
      <span
        class="MuiToggleButton-label"
      >
        Sydney
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Brisbane"
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-19 MuiToggleButtonGroup-grouped toggle-button-group-grouped-16 toggle-button-group-grouped-18 MuiToggleButtonGroup-groupedHorizontal Mui-selected toggle-button-selected-20"
      data-id="brsb"
      tabindex="0"
      type="button"
      value="brsb"
    >
      <span
        class="MuiToggleButton-label"
      >
        Brisbane
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Perth"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-19 MuiToggleButtonGroup-grouped toggle-button-group-grouped-16 toggle-button-group-grouped-18 MuiToggleButtonGroup-groupedHorizontal"
      data-id="prth"
      tabindex="0"
      type="button"
      value="prth"
    >
      <span
        class="MuiToggleButton-label"
      >
        Perth
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Melbourne"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-19 MuiToggleButtonGroup-grouped toggle-button-group-grouped-16 toggle-button-group-grouped-18 MuiToggleButtonGroup-groupedHorizontal"
      data-id="melb"
      tabindex="0"
      type="button"
      value="melb"
    >
      <span
        class="MuiToggleButton-label"
      >
        Melbourne
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`It renders with object[] options 1`] = `
<div>
  <div
    aria-label="Australian city"
    class="MuiToggleButtonGroup-root toggle-button-group-root-8 toggle-button-group-root-10"
    data-id="Australian city"
    id="Australian city"
    role="group"
  >
    <button
      aria-label="Sydney"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-12 MuiToggleButtonGroup-grouped toggle-button-group-grouped-9 toggle-button-group-grouped-11 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-9 toggle-button-group-grouped-11"
      data-id="sydn"
      tabindex="0"
      type="button"
      value="sydn"
    >
      <span
        class="MuiToggleButton-label"
      >
        Sydney
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Brisbane"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-12 MuiToggleButtonGroup-grouped toggle-button-group-grouped-9 toggle-button-group-grouped-11 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-9 toggle-button-group-grouped-11"
      data-id="brsb"
      tabindex="0"
      type="button"
      value="brsb"
    >
      <span
        class="MuiToggleButton-label"
      >
        Brisbane
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Perth"
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-12 MuiToggleButtonGroup-grouped toggle-button-group-grouped-9 toggle-button-group-grouped-11 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-9 toggle-button-group-grouped-11 Mui-selected toggle-button-selected-13"
      data-id="prth"
      tabindex="0"
      type="button"
      value="prth"
    >
      <span
        class="MuiToggleButton-label"
      >
        Perth
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Melbourne"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-12 MuiToggleButtonGroup-grouped toggle-button-group-grouped-9 toggle-button-group-grouped-11 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-9 toggle-button-group-grouped-11"
      data-id="melb"
      tabindex="0"
      type="button"
      value="melb"
    >
      <span
        class="MuiToggleButton-label"
      >
        Melbourne
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`It renders with string[] options 1`] = `
<div>
  <div
    aria-label="Australian city"
    class="MuiToggleButtonGroup-root toggle-button-group-root-1 toggle-button-group-root-3"
    data-id="Australian city"
    id="Australian city"
    role="group"
  >
    <button
      aria-label="Sydney"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-5 MuiToggleButtonGroup-grouped toggle-button-group-grouped-2 toggle-button-group-grouped-4 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-2 toggle-button-group-grouped-4"
      data-id="Sydney"
      tabindex="0"
      type="button"
      value="Sydney"
    >
      <span
        class="MuiToggleButton-label"
      >
        Sydney
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Brisbane"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-5 MuiToggleButtonGroup-grouped toggle-button-group-grouped-2 toggle-button-group-grouped-4 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-2 toggle-button-group-grouped-4"
      data-id="Brisbane"
      tabindex="0"
      type="button"
      value="Brisbane"
    >
      <span
        class="MuiToggleButton-label"
      >
        Brisbane
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Perth"
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-5 MuiToggleButtonGroup-grouped toggle-button-group-grouped-2 toggle-button-group-grouped-4 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-2 toggle-button-group-grouped-4 Mui-selected toggle-button-selected-6"
      data-id="Perth"
      tabindex="0"
      type="button"
      value="Perth"
    >
      <span
        class="MuiToggleButton-label"
      >
        Perth
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Melbourne"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root toggle-button-root-5 MuiToggleButtonGroup-grouped toggle-button-group-grouped-2 toggle-button-group-grouped-4 MuiToggleButtonGroup-groupedHorizontal toggle-button-group-grouped-2 toggle-button-group-grouped-4"
      data-id="Melbourne"
      tabindex="0"
      type="button"
      value="Melbourne"
    >
      <span
        class="MuiToggleButton-label"
      >
        Melbourne
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
